README.md

```diff
@@ -1,407 +1,436 @@
+| scroll up in preview window      | `<C-u>`        |
+| scroll down in preview window    | `<C-d>`        |
+| close telescope                  | `<C-c>`        |
+| close telescope (in normal mode) | `<Esc>`        |
 
-```
+To see the full list of mappings, check out `lua/telescope/mappings.lua` and
+the `default_mappings` table.  
 
-A full example:
+To change default mapping globally, then change default->mappings dict
+<!-- TODO should be in the wiki -->
 
 ```lua
 local actions = require('telescope.actions')
 
-require('telescope').setup {
+-- If you want your function to run after another action you should define it as follows
+local test_action = actions._transform_action(function(prompt_bufnr)
+  print("This function ran after another action. Prompt_bufnr: " .. prompt_bufnr)
+  -- Enter your function logic here. You can take inspiration from lua/telescope/actions.lua
+end)
+
+require('telescope').setup{
   defaults = {
     mappings = {
       i = {
-        -- Disable the default <c-x> mapping
+        -- To disable a keymap, put [map] = false
+        -- So, to not map "<C-n>", just put
         ["<c-x>"] = false,
-
-        -- Create a new <c-s> mapping
-        ["<c-s>"] = actions.goto_file_selection_split,
+        -- Otherwise, just set the mapping to the function that you want it to be.
+        ["<C-i>"] = actions.goto_file_selection_split,
+        -- Add up multiple actions
+        ["<CR>"] = actions.goto_file_selection_edit + actions.center,
+        -- You can perform as many actions in a row as you like
+        ["<CR>"] = actions.goto_file_selection_edit + actions.center + test_action,
+      },
+      n = {
+        ["<esc>"] = actions.close
       },
     },
   }
 }
-```
 
-To override only SOME of the default mappings, you can use the `attach_mappings` key in the `setup` table. For example:
-
-```lua
-function my_custom_picker(results)
-  pickers.new(opts, {
-    prompt_title = 'Custom Picker',
-    finder = finders.new_table(results),
-    sorter = sorters.fuzzy_with_index_bias(),
-    attach_mappings = function(_, map)
-      -- Map "<CR>" in insert mode to the funciton, actions.set_command_line
-      map('i', '<CR>', actions.set_command_line)
-
-      return true
-    end,
-  }):find()
+-- You can also define your own functions, which then can be mapped to a key
+local function test_action(prompt_bufnr)
+  print("Action was attached with prompt_bufnr: ", prompt_bufnr)
+-- Enter your function logic here. You can take inspiration from lua/telescope/actions.lua
 end
+["<C-i>"] = test_action,
+
+-- If you want your function to run after another action you should define it as follows
+local test_action = actions._transform_action(function(prompt_bufnr)
+  print("This function ran after another action. Prompt_bufnr: " .. prompt_bufnr)
+-- Enter your function logic here. You can take inspiration from lua/telescope/actions.lua
+end)
+["<C-i>"] = actions.goto_file_selection_split + test_action
```

Comment:
We kinda already have this one in the example above.

Reply:
@Conni2461 What do you mean?

Reply:
The lower part 274-279 is already in the example above and the upper part is how you can define an action as well but i think we can just advertise the correct way to define one. In that way people will always create actions that can be concatenated.

Reply:
Sorry @Conni2461 still couldn't understand where to look :laughing: can you please comment on the newest version of the readme. I will mark this as resolved for now.